backend/ethereum/channel/conclude.go

```diff
@@ -119,6 +133,30 @@ func (a *Adjudicator) isConcluded(ctx context.Context, sub *subscription.EventSu
 	return false, errors.WithMessage(<-subErr, "reading past events")
 }
 
+// isForceExecuted returns whether a channel is already concluded.
+func (a *Adjudicator) isForceExecuted(ctx context.Context, c channel.ID) (bool, error) {
+	sub, err := subscription.NewEventSub(ctx, a.ContractBackend, a.bound, updateEventType(c), startBlockOffset)
+	if err != nil {
+		return false, errors.WithMessage(err, "subscribing")
+	}
+	defer sub.Close()
+	events := make(chan *subscription.Event, 10)
+	subErr := make(chan error, 1)
+	// Write the events into events.
+	go func() {
+		defer close(events)
+		subErr <- sub.ReadPast(ctx, events)
```

Comment:
I think in theory this would need a sub-ctx as context, to ensure that it gets cancelled on function exit.  
In practice there will probably not be enough events anyway.

Reply:
ok, let's use a sub-context then.